modules/ObjectControl/src/objectcontrol.cpp

```diff
@@ -690,6 +714,107 @@ void ObjectControl::startScenario() {
 	}
 }
 
+void ObjectControl::resetTest() {
+	// Check if we are already resetting
+	if (this->isResetting) {
+		RCLCPP_INFO(get_logger(), "Test already resetting");
+		return;
+	}
+	RCLCPP_INFO(get_logger(), "Resetting test by offering return trajectories");
+	// Request a new return trajectory for each object
+	for (auto& id : getVehicleIDs()) {
+		this->setObjectTrajectory(id);
+	}
+	this->isResetting = true;
+}
+
+void ObjectControl::reloadScenarioTrajectories() {
+	// Check if we already have loaded the scenario trajectories
+	if (!this->isResetting) {
+		RCLCPP_INFO(get_logger(), "Scenario trajectories already loaded");
+		return;
+	}
+	RCLCPP_INFO(get_logger(), "Reloading scenario trajectories");
+	// Request the scenario trajectory for each object
+	for (auto& id : getVehicleIDs()) {
+		this->setObjectTrajectory(id);
+	}
+	this->isResetting = false;
+}
+
+void ObjectControl::updateTrajectoryGUI(uint32_t id){
```

Comment:
Could be called e.g. republishTrajectories. The GUI is just one of many potential "listeners" on the topic.

Reply:
Renamed in new commit